### DISCUSSION TITLE
How to run a code inside canvas document while deleting the component?

### DISCUSSION THREAD

**Discussion by aswin-g2** [GrapesJS/grapesjs#5601] [Q&A]
The 'script' option below is executed when the component is added. Is there any way to execute additional code while the component is being removed? 

Note that the script need to be executed in canvas document.

Any help would be appreciated. 

```js
editor.Components.addType('slider', {
    model: {
        defaults: {
            script: function () {
                // `this` is bound to the component element
            }
        }
    }
});
```

**Comment by artf** [comment]
I guess you can make use of `removed()` [hook](https://grapesjs.com/docs/modules/Components.html#lifecycle-hooks) in your component.

**Reply by aswin-g2** [reply]
@artf I have already tried, but it runs the code in the main window, not inside the canvas. As I mentioned in the discussion, I want the code to be executed inside the canvas document.

**Reply by artf** [reply]
You can get the document context from the DOM element if you need it (eg. `editor.getSelected().getEl().ownerDocument`).
